BlocksForAll/ColorModViewController.swift

```diff
@@ -15,29 +15,160 @@ class ColorModViewController: UIViewController{
+                button.isSelected = false
+                button.layer.borderWidth = 0
+            } else {
+                deselectAll()
+                button.isSelected = true
+                button.layer.borderWidth = 5
+                button.layer.borderColor = #colorLiteral(red: 0.004859850742, green: 0.09608627111, blue: 0.5749928951, alpha: 1)
+            }
+        }
     }
+    
     @IBAction func yellowButtonPressed(_ sender: UIButton) {
         colorSelected = "yellow"
+        //Border when button is selected
+        if let button = sender as? UIButton {
+            if button.isSelected {
+                button.isSelected = false
+                button.layer.borderWidth = 0
+            } else {
+                deselectAll()
+                button.isSelected = true
+                button.layer.borderWidth = 5
+                button.layer.borderColor = #colorLiteral(red: 0.004859850742, green: 0.09608627111, blue: 0.5749928951, alpha: 1)
+            }
+        }
     }
+    
     @IBAction func greenButtonPressed(_ sender: UIButton) {
         colorSelected = "green"
+        //Border when button is selected
+        if let button = sender as? UIButton {
+            if button.isSelected {
+                button.isSelected = false
+                button.layer.borderWidth = 0
+            } else {
+                deselectAll()
+                button.isSelected = true
+                button.layer.borderWidth = 5
+                button.layer.borderColor = #colorLiteral(red: 0.004859850742, green: 0.09608627111, blue: 0.5749928951, alpha: 1)
+            }
+        }
     }
+    
     @IBAction func blueButtonPressed(_ sender: UIButton) {
         colorSelected = "blue"
+        //Border when button is selected
+        if let button = sender as? UIButton {
+            if button.isSelected {
+                button.isSelected = false
+                button.layer.borderWidth = 0
+            } else {
+                deselectAll()
+                button.isSelected = true
+                button.layer.borderWidth = 5
+                button.layer.borderColor = #colorLiteral(red: 0.004859850742, green: 0.09608627111, blue: 0.5749928951, alpha: 1)
+            }
+        }
     }
+    
     @IBAction func purpleButtonPressed(_ sender: UIButton) {
         colorSelected = "purple"
+        //Border when button is selected
+        if let button = sender as? UIButton {
+            if button.isSelected {
+                button.isSelected = false
+                button.layer.borderWidth = 0
+            } else {
+                deselectAll()
+                button.isSelected = true
+                button.layer.borderWidth = 5
+                button.layer.borderColor = #colorLiteral(red: 0.004859850742, green: 0.09608627111, blue: 0.5749928951, alpha: 1)
+            }
+        }
     }
+    
+    
     @IBAction func whiteButtonPressed(_ sender: UIButton) {
         colorSelected = "white"
```

Comment:
lines 157 to 171
       //Border when button is selected
         
         if let button = sender as? UIButton {
             if button.isSelected {
                 button.isSelected = false
                 button.layer.borderWidth = 0

 
              } else {
                 deselectAll()
                 button.isSelected = true
                 button.layer.borderWidth = 5
                 button.layer.borderColor = #colorLiteral(red: 0.004859850742, green: 0.09608627111, blue: 0.5749928951, alpha: 1)
             }
         }